### PR TITLE
Support catalog register new schema version

### LINKED ISSUE #1060: Support `catalog` register new schema version
Enhance `CatalogHandler` API to support register new schema version.

Implement `catalog-schema-registry` register method.
No need to support `apicurio` catalog yet.

### PR DESCRIPTION
**Pull request by akrambek**
## Description

Support catalog register new schema version.

Fixes #1060

### CODE REVIEW

File: runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 public class SchemaRegistryCatalogHandler implements CatalogHandler
 {
     private static final String SUBJECT_VERSION_PATH = "/subjects/{0}/versions/{1}";
+    private static final String SUBJECT_PATH = "subjects/{0}/versions";
```
Comment (jfallows) [inline runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java:45]: Missing leading slash?

File: runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/catalog/CatalogHandler.java [hunk 2]
```diff
@@ -61,6 +62,13 @@ int accept(
             ValueConsumer next);
     }
 
+    default int register(
+        String subject,
+        String schema)
+    {
+        return NO_VERSION_ID;
+    }
+
```
Comment (jfallows) [inline runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/catalog/CatalogHandler.java:71]: Excellent.

File: specs/catalog-schema-registry.spec/src/main/scripts/io/aklivity/zilla/specs/catalog/schema/registry/config/register/zilla.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      url: http://localhost:8081
+bindings:
+  net0:
+    type: test
+    kind: server
+    options:
+      catalog:
+        catalog0:
+          - subject: items-snapshots-value
+            record: |
```
Comment (jfallows) [inline specs/catalog-schema-registry.spec/src/main/scripts/io/aklivity/zilla/specs/catalog/schema/registry/config/register/zilla.yaml:31]: `record` is something else, this should be `schema`.

File: runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/binding/TestBindingFactory.java [hunk 4]
```diff
@@ -347,7 +347,11 @@ else if (assertion.schema != null && !assertion.schema.equals(schema))
                 }
                 else
                 {
-                    if (catalog.subject != null && catalog.version != null)
+                    if (catalog.subject != null && catalog.record != null)
+                    {
+                        handler.register(catalog.subject, catalog.record);
+                    }
```
Comment (jfallows) [inline runtime/engine/src/test/java/io/aklivity/zilla/runtime/engine/test/internal/binding/TestBindingFactory.java:353]:
```suggestion
                    if (catalog.subject != null && catalog.schema != null)
                    {
                        handler.register(catalog.subject, catalog.schema);
                    }
```

**Review by jfallows — CHANGES_REQUESTED** [review]

File: runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java (hunk 1 above)
Comment (akrambek) [reply runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java:45]: Nope, no slash required. Spec is also correct `accept "http://localhost:8081/subjects/items-snapshots-value/versions"`

Comment (jfallows) [reply runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java:45]: Why does `SUBJECT_VERSION_PATH` start with `/subjects/...` but `SUBJECT_PATH` starts with `subjects/...`?
Seems like these should be consistent, and paths tend to start with `/`, no?

Comment (akrambek) [reply runtime/catalog-schema-registry/src/main/java/io/aklivity/zilla/runtime/catalog/schema/registry/internal/handler/SchemaRegistryCatalogHandler.java:45]: oh I see what you mean I got it wrong yep :)

**Review by jfallows — DISMISSED** [review]

**Review by jfallows — APPROVED** [review]

**Pull request merged**